File: swift/problem205.swift
class Solution {
    func isIsomorphic(_ s: String, _ t: String) -> Bool {
        if (s.count != t.count) {
            return false
        }
        
        let s = s.map{$0}; let t = t.map{$0}
        var mapping = [Character:Character]()
        
        for i in 0..<s.count {
            let sTmp = s[i]; let tTmp = t[i]
            if (mapping[sTmp] != nil) {
                if (tTmp != mapping[sTmp]!) {
                    return false
                }
            } else if mapping.values.contains(tTmp) {
                return false
            } else {
                mapping[sTmp] = tTmp
            }
        }
        return true
    }
}

File: python/problem205.py
class Solution:
    def isIsomorphic(self, s: 'str', t: 'str') -> 'bool':
        if len(set(s)) != len(set(t)):
            return False
        
        def strtoint(self,s):
            se = set()
            dic = {}
            li = []
            k = 0
            for i in range(len(s)):
                if s[i] in se:
                    li.append(dic[s[i]]) 

                else:
                    se.add(s[i])
                    dic[s[i]]= k
                    li.append(k) 
                    k+=1
            return li
        return strtoint(self,s) == strtoint(self,t)
